### PR TITLE
Refactor use global id for processing

### PR DESCRIPTION
**Pull request by KimballCai**
step 1: format files without logical change into the previous version
step 2: review the code

### CODE REVIEW

**Comment by KimballCai** [comment]
have fixed the import order problem

**Comment by hugy718** [comment]
I will do one round of checking to remove further code formatting issues tmr morning.

**Review by hugy718 — COMMENTED** [review]
Code has become much cleaner now thanks to the improved abstraction. Kudos to @Zrealshadow,

Pushed a commit to clean and format the codes as i reviewed it.

I have not read about the iceberg files (there are 5), and those cool model files to be removed. The cubemeta is indeed broken, likely due to the encoding has been changed in io. I will fix that later.

I have written comments where I have doubts.

File: cool-core/src/main/java/com/nus/cool/core/io/compression/ZIntCompressor.java [hunk 1]
```diff
@@ -104,11 +106,9 @@ public int compress(int[] src, int srcOff, int srcLen, byte[] dest, int destOff,
         case 4:
           buffer.putInt(src[i]);
           break;
-        default:
```
[the remark this answers is not in the document]
Comment (hugy718) [reply cool-core/src/main/java/com/nus/cool/core/io/compression/ZIntCompressor.java:107]: Oh, it was added by me during code cleaning to fix the warning...

**Comment by hugy718** [comment]
cubemeta is working now.

**Comment by KimballCai** [comment]
Please add more instructions for this branch.

**Comment by KimballCai** [comment]
It seems that the code has already been rebased into dev.

**Pull request merged**